### PR TITLE
Feature/#242 create endpoint cancelling appointment specialist

### LINKED ISSUE #242: SP: 5 Create endpoint for cancelling appointment for Specialist

### PR DESCRIPTION
**Pull request by Misha86**
Create an endpoint for cancelling appointment specialists.
Added sending an email notification.

### CODE REVIEW

File: beauty/sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,21 @@
+sonar-scanner \
```
Comment (OleksiiDatsiuk) [inline beauty/sonar-project.properties:1]: Do we have sonarcube configured for our project?

**Review by OleksiiDatsiuk — APPROVED** [review]

File: beauty/api/views/order_views.py [hunk 2]
```diff
@@ -104,6 +103,21 @@ def get_object(self):
...
+        super().put(request, *args, **kwargs)
+        order = self.get_object()
+        authenticated_user = request.user
+        user = order.customer if authenticated_user == order.specialist else order.specialist
+        context = {"order": order, "user": authenticated_user}
+
+        CancelOrderEmail(request, context).send([user.email])
+
+        logger.info(f"{order}: canceling email was sent to the {user.get_full_name()}")
+
```
Comment (OleksiiDatsiuk) [inline beauty/api/views/order_views.py:117]: Also, think that we need add reason of cancellation as a required field and check whether its included into put request data and to make sure that its length != 0

File: beauty/sonar-project.properties (hunk 1 above)
Comment (Misha86) [reply beauty/sonar-project.properties:1]: Sorry, I thought it is in the .gitignore.

File: beauty/api/views/order_views.py (hunk 2 above)
Comment (Misha86) [reply beauty/api/views/order_views.py:117]: Reason field is a requirement in our serializer and it can not be blank

Comment (OleksiiDatsiuk) [reply beauty/api/views/order_views.py:117]: I am not sure if it is a good idea to let it be blank, I mean, for example, every business owner wants to know the reason why one of the orders was cancelled

Comment (Misha86) [reply beauty/api/views/order_views.py:117]: Sorry, can not be blank

Comment (OleksiiDatsiuk) [reply beauty/api/views/order_views.py:117]: I mean, yeah, this is not a case for customers, but on the other hand, it is required for specialist to specify cancellation reason

Comment (OleksiiDatsiuk) [reply beauty/api/views/order_views.py:117]: > Sorry, can not be blank

Oh, than thats it, thx a lot

Comment (Misha86) [reply beauty/api/views/order_views.py:117]: If a field is a requirement it can not be blank.

**Review by Serhii-Voloshyn — APPROVED** [review]

**Review by therobotisnotatoy — APPROVED** [review]

**Pull request merged**